src/app/containers/MostRead/utilities/processMostRead.js

```diff
@@ -60,48 +57,44 @@ const mostReadItems = ({ data, isAmp, numberOfItems, service }) => {
   if (!data) {
     return null;
   }
-  const records = pathOr([], ['records'], data);
 
-  // The ARES test endpoint for most read renders fixture data, so the data is stale
-  const isTest = process.env.SIMORGH_APP_ENV === 'test';
+  const filteredData = filterPopularStaleData({
+    data,
+    isAmp,
+    service,
+    popularType: 'mostRead',
+  });
+
+  if (!filteredData) {
+    return null;
+  }
 
-  // Do not show most read if lastRecordUpdated is greater than 60min as this means PopAPI has failed twice
-  // in succession. This suggests ATI may be having issues, hence risk of stale data.
-  if (isTest || mostReadRecordIsFresh(data.lastRecordTimeStamp)) {
-    const items = [];
-    // eslint-disable-next-line no-restricted-syntax
-    for (const record of records) {
-      const mostReadItemData =
-        record.promo.type === 'optimo'
-          ? getOptimoItemData(record)
-          : getCpsItemData(record);
-      const { href, title } = mostReadItemData;
+  const records = pathOr([], ['records'], filteredData);
 
-      if (href && title) {
-        items.push(mostReadItemData);
-      } else {
-        logger.warn(MOST_READ_DATA_INCOMPLETE, {
-          service,
-          title,
-          url: href,
-        });
-      }
+  const items = [];
+  // eslint-disable-next-line no-restricted-syntax
+  for (const record of records) {
+    const mostReadItemData =
+      record.promo.type === 'optimo'
+        ? getOptimoItemData(record)
+        : getCpsItemData(record);
+    const { href, title } = mostReadItemData;
 
-      if (items.length === numberOfItems) {
-        break;
-      }
+    if (href && title) {
+      items.push(mostReadItemData);
+    } else {
+      logger.warn(MOST_READ_DATA_INCOMPLETE, {
+        service,
+        title,
+        url: href,
```

Comment:
I know this is not code that was written by you originally, but would it be useful to know which fields are incomplete? If either title or url are undefined, the value simply won't appear in the logs.

Reply:
It will be useful, but I think it's out of the scope of this PR. Maybe we can open a new issue for that